plugin-server/src/cdp/utils.ts

```diff
@@ -460,3 +462,122 @@ export function buildExportedFunctionInvoker(
 export function isLegacyPluginHogFunction(hogFunction: HogFunctionType): boolean {
     return hogFunction.template_id?.startsWith('plugin-') ?? false
 }
+
+/**
+ * Shared utility to check if an event matches the filters of a HogFunction.
+ * Used by both the HogExecutorService (for destinations) and HogTransformerService (for transformations).
+ *
+ * @param hogFunction The function to check filters for
+ * @param filters The function filters
+ * @param filterGlobals The globals to use for filter evaluation
+ * @param telemetryMatcher Optional telemetry matcher for monitoring
+ * @param eventUuid Optional event UUID for error logging
+ * @returns {boolean} Whether the filters pass (true) or not (false)
+ */
+export function checkHogFunctionFilters(
+    hogFunction: HogFunctionType,
+    filters: HogFunctionType['filters'],
+    filterGlobals: HogFunctionFilterGlobals,
+    options?: {
+        telemetryMatcher?: (teamId: number) => boolean
+        eventUuid?: string
+        metricsCollector?: (metric: HogFunctionAppMetric) => void
+        logsCollector?: (log: HogFunctionInvocationLogEntry) => void
+        durationObserver?: (duration: number) => void
```

Comment:
These should all just be returned as values. Side effect options here feel super confusing to me...